WordPress/Sniffs/Theme/DeprecatedWPFunctionsSniff.php

```diff
@@ -0,0 +1,341 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     https://make.wordpress.org/core/handbook/best-practices/coding-standards/
+ */
+
+/**
+ * WordPress_Sniffs_Theme_DeprecatedWPFunctionsSniff.
+ *
+ * Forbids the use of deprecated WordPress functions in themes.
+ * Note that the array must be updated every major WordPress release.
+ * Note that $current_wp_version must be updated every major WordPress release.
+ * See includes/deprecated.
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   khacoder
+ */
+class WordPress_Sniffs_Theme_DeprecatedWPFunctionsSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_STRING,
+			T_EVAL,
+		);
+
+	} // end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token
+	 *                                        in the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+
+		$tokens = $phpcsFile->getTokens();
+		$token  = $tokens[ $stackPtr ];
+
+		/*
+		 * current_wp_version is used to get the cut off from WARNING to ERROR.
+		 * The current cut off is 3 versions back, meaning WARNINGS for the
+		 * latest 4 versions and ERRORS before that.
+		 */
+		$current_wp_version = 4.6;
+		$cut_off = $current_wp_version - 0.3;
+
+		$checks = array(
+			// start wp-includes deprecated.
+			array( 'get_postdata' => 'get_post()', '1.5.1' ),
```

Comment:
It would be good to add spacing in between the arguments so that it is a bit easier to read. e.g.

```php
array( 'get_postdata' => 'get_post()', '1.5.1' ),
array( 'start_wp'     => 'the Loop',   '1.5' ),
```